docs_src/microservices/device/profile/Ch-DeviceProfile.md

```diff
@@ -1,6 +1,93 @@
+for identification and a description for informational purposes.
+
+The device service allows access to deviceResources via its `device`
+REST endpoint.
+
+The `Attributes` in a deviceResource are the device-service-specific parameters
+required to access the particular value. Each device service implementation
+will have its own set of named values that are required here, for example a
+BACnet device service may need an Object Identifier and a Property Identifier
+whereas a Bluetooth device service could use a UUID to identify a value.
+
+The `Properties` of a deviceResource describe the value. Each device resource
+is given two properties, `value` and `units`. The following fields are
+available in a property:
+
+* type - Required. The data type of the value. Supported types are `bool`,
+`int8` - `int64`, `uint8` - `uint64`, `float32`, `float64`, `string`, `binary`
+and arrays of the primitive types (ints, floats, bool). Arrays are specified
+as eg. `float32array`, `boolarray` etc.
+* readWrite - `R`, `RW`, or `W` indicating whether the value is readable or
+writable.
+* defaultValue - a value used for PUT requests which do not specify one.
+* base - a value to be raised to the power of the raw reading before it is returned.
+* scale - a factor by which to multiply a reading before it is returned.
+* offset - a value to be added to a reading before it is returned.
+* mask - a binary mask which will be applied to an integer reading.
+* shift - a number of bits by which an integer reading will be shifted right.
+
+The processing defined by base, scale, offset, mask and shift is applied in
+that order. This is done within the SDK. A reverse transformation is applied
+by the SDK to incoming data on set operations (NB mask transforms on set are NYI)
+
+The `units` property is used to indicate the units of the value, eg Amperes,
+degrees C, etc. It should have a `type` of `String`, `readWrite` of `R`
+indicating read-only, and a `defaultValue` that specifies the units.
+
+DeviceCommands
+--------------
+
+DeviceCommands define access to reads and writes for multiple simultaneous
+values. Each named deviceCommand should contain a number of get and/or set
+`resourceOperations`, describing the read or write respectively.
+
+A resourceOperation consists of the following properties:
+
+* index - a number, used to define an order in which the resource is processed.
+* operation - `get` or `set`. Ignored in this implementation, mixing of get
+and set operations is not supported.
+* deviceResource - the name of the deviceResource to access.
+* parameter - optional, a value that will be used if a PUT request does not
+specify one.
+* mappings - optional, allows readings of String type to be re-mapped.
+
+The device service allows access to deviceCommands via the `device` REST
+endpoint. If a deviceCommand and deviceResource have the same name, it will be
+the deviceCommand which is available.
+
+CoreCommands
```

Comment:
I think it would be helpful to specify where/why there might be a resource operation but not a command.